### PR TITLE
Command line arguments

### PR DESCRIPTION
**Pull request by shakram02**
Output will be printed to **${name}_out/main** only

for #35

### CODE REVIEW

**Comment by sushant94** [comment]
Issue #35 says that in addition to having separate per output file, we need one all functions in it. Sorry, I guess that wasn't clear as there was no description. We would still like to keep the per-function output file too.

**Comment by shakram02** [comment]
okay, I'll restore the multiple files

**Comment by shakram02** [comment]
@sushant94 done

File: minidec/main.rs [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+use std::io::Write;
+use std::path::{Path, PathBuf};
 
 fn main() {
+    // Creates cmd args settings then gets the matching params
+    let matches = cli::create_args().get_matches();
+
     let mut dir;
-    let mut r2 = R2::new::<String>(None).expect("Unable to open r2");
+    let mut r2 = R2::new::<String>(Some("/home/ahmed/radare2/radeco-lib/ex-bins/hello-linux-x86_64"
```
Comment (sushant94) [inline minidec/main.rs]: I don't think this should be hardcoded.

`None` is passed when you want `r2::R2` to use the currently opened file inside r2.
If you want to open a specific file to analyze, you could have it as another argument.

File: Cargo.toml [hunk 2]
```diff
@@ -24,6 +24,7 @@ num = "*"
 rustc-serialize = "*"
 lazy_static = "*"
 log = "*"
+clap="*"
```
Comment (sushant94) [inline Cargo.toml]: I'm fine with `clap` but you could look into using `docopt.rs` too.

File: minidec/main.rs [hunk 3]
```diff
@@ -37,7 +44,44 @@ fn main() {
...
+    // ?? : rmod.functions returns (&u64,RFunction) the names from cmd args are matched with the
+    // name field in RFunction, the name is then not needed to access the function
+    // info, only the u64 address is needed, so the unnecessary info are filtered out
+    let matched_func_vec: Vec<(u64, &String)> = rmod.functions
+            .iter()
+            .map(|(fn_addr, rfn)| (*fn_addr, &rfn.name)) // Tuple of u64 and &String
+            .filter(|&(_, &ref fn_name): &(u64, &String)| { // Types are kept just for clearence
+                
+                // The user didn't specify any args, match all
+                if requested_functions.len() == 0{return true;}
```
Comment (sushant94) [inline minidec/main.rs]: Better to have this check outside and restructure this loop.

File: minidec/main.rs (hunk 1 above)
Comment (shakram02) [inline minidec/main.rs]: oh yes I forgot to remove it, I'll fix it

File: Cargo.toml (hunk 2 above)
Comment (shakram02) [inline Cargo.toml]: okay, I'll change it, but why do you prefer `docopt` does it have any extra features over `clap` ?

**Comment by shakram02** [comment]
@sushant94 I updated the commit as requested

**Comment by sushant94** [comment]
Alright, this seems good to me. Merging. Thanks!

**Pull request merged**